EventListener/AbstractRuleSubscriber.php

```diff
@@ -54,4 +54,16 @@ protected function matchRule(Request $request, Response $response)
 
         return false;
     }
+
+    /**
+     * Decide whether to even look for matching rules with the current request.
+     *
+     * @param Request $request
+     *
+     * @return bool True if the request is safe and headers can be set
+     */
+    protected function isRequestSafe(Request $request)
```

Comment:
can we rename this method to `isRequestCacheable`? this is what we semantically want to know.